app-testsuite/cypress/integration/full_spec.js

```diff
@@ -10,7 +10,16 @@ describe('counter', () => {
 describe('secret', () => {
   it('gets env var', () => {
     cy.visit('/');
-    cy.get('.secret').contains('shhh');
+    cy.get('.secret').should('have.text', 'shhhh');
+  });
+});
+
+describe('app secret', () => {
+  it('gets env var set from secrets API', () => {
+    cy.visit('/');
+    if (cy.get('.runningOn').invoke('text') === 'realm') {
```

Comment:
This won't work as cypress is not actually synchronous, I think [then](https://docs.cypress.io/api/commands/then.html#Syntax) should work. Cypress API is actually async, but not in the `async/await` way (I think this was a good source https://medium.com/@NicholasBoll/cypress-io-using-async-and-await-4034e9bab207).